baseplate/ratelimit/backends/memcache.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from .base import RateLimitBackend
+from ...context import ContextFactory
+from ...context.memcache import MonitoredMemcacheConnection
+
+
+class MemcacheRateLimitBackendContextFactory(ContextFactory):
+    """MemcacheRateLimitBackend context factory
+
+    :param memcache_pool: An instance of
+        :py:class:`~pymemcache.client.base.PooledClient`
+
+    """
+
+    def __init__(self, memcache_pool):
+        self.memcache_pool = memcache_pool
+
+    def make_object_for_context(self, name, server_span):
+        memcache = MonitoredMemcacheConnection(name, server_span, self.memcache_pool)
```

Comment:
i just wanted to raise these classes peeking into the guts of the memcached/redis context factories. it's probably ok since this is all in baseplate, but it does feel a bit coupled. would it be insane to construct a context factory in `__init__` and call its `make_object_for_context` here to get a monitored connection? too weird with a fake context variable?

Reply:
I think this sounds reasonable?

Reply:
The approach that you describe does seem like better design, but the fake context is kinda weird. I'm okay with this as it is, but I'll implement your advice if you feel strongly about it. I could go either way.

Reply:
Didn't see Andrew's comment before making mine. I'll change it to use `make_object_for_context ` since I'm the odd one out. 👍